### PR TITLE
Install software as used within elgentos

### PR DESCRIPTION
**Pull request by janmartenjongerius**
### Changes

- Google Chrome is now optional

### Additions

- The AWS CLI command `aws` is installed using Docker
- Added Brave Browser
- Added Firefox
- Added Dnsmasq
- Added Dnsmasq configuration for Docker Compose Development
- Added ssg-js

### CODE REVIEW

File: .zshrc [hunk 1]
```diff
@@ -18,7 +18,15 @@ plugins=(
...
+    hash -d "$(basename "$workspace")=$workspace";
+done
+
 source $ZSH/oh-my-zsh.sh
 
 # Profile generated for docker-compose-development.
 [ -f "$HOME/.zshrc-development" ] && source "$HOME/.zshrc-development"
+
+# Run AWS CLI through Docker.
+alias aws='docker run --rm -it -v ~/.aws:/root/.aws -v $(pwd):/aws amazon/aws-cli'
```
Comment (janmartenjongerius) [inline .zshrc]: @JeroenBoersma is there anything you personally added on top of this?

Comment (JeroenBoersma) [reply .zshrc]: Yes I've... 

Here you go!

Comment (JeroenBoersma) [reply .zshrc]: https://gist.github.com/JeroenBoersma/87e29fd4aa06ec42216c80a6e3649fa5

**Comment by janmartenjongerius** [comment]
@JeroenBoersma the following changes are made

- The `aws` alias has been turned into a local installation of your AWS gist, allowing developers to `git pull` for updates, as you announce them.
- Added Dnsmasq
- Added Dnsmasq configuration for Docker Compose Development, which allows DNS resolution of `*.localhost` (or whatever the configured `$DOMAINSUFFIX` is set at within the development repo) within Docker containers.

**Comment by janmartenjongerius** [comment]
I just found out the nameserver configuration does not survive a reboot. Will have to look into that.

File: Makefile [hunk 2]
```diff
@@ -351,12 +394,149 @@ teamviewer: | $(TEAMVIEWER)
...
+
+$(DNSMASQ): | $(BASH) $(UFW)
+	$(BASH) -c '[ -f /.dockerenv ] || sudo $(SYSTEMCTL) disable --now systemd-resolved'
+	$(BASH) -c '[ -f /.dockerenv ] || sudo rm -f /etc/resolv.conf'
+	echo 'nameserver 127.0.0.1' | sudo tee    /etc/resolv.conf
+	echo 'nameserver 1.1.1.1'   | sudo tee -a /etc/resolv.conf
+	echo 'nameserver 9.9.9.9'   | sudo tee -a /etc/resolv.conf
+	sudo mkdir -p /etc/NetworkManager/conf.d
+	echo "[main]\ndns=none\nrc-manager=unmanaged" | sudo tee /etc/NetworkManager/conf.d/dnsmasq.conf
+	sudo service --status-all | awk '{ print $$4 }' | grep -qv network-manager || sudo service network-manager restart
```
Comment (janmartenjongerius) [inline Makefile]: Together with Arjen, I observed this command is skipped when it shouldn't be skipped.

File: .zshrc [hunk 3]
```diff
@@ -18,6 +18,11 @@ plugins=(
 # Make all GIT repos into named directories.
 for repo in ~/git/*; do hash -d "$(basename "$repo")=$repo"; done
 
+# Make all workspaces into named directories.
+[ -d ~/development/workspace ] && for workspace in ~/development/workspace/*; do
```
Comment (JeroenBoersma) [inline .zshrc:22]: you can put your workspace anywhere with `dev volume workspace`

there is a command to retrieve the `workspace` location `dev workspace` which returns it as a string.

File: Makefile [hunk 4]
```diff
@@ -351,12 +397,153 @@ teamviewer: | $(TEAMVIEWER)
...
+	brave \
+	docker-compose-development-dnsmasq \
+	firefox \
+	gimp \
+	google-chrome \
+	symlinks \
+	node \
+	slack \
+	ssg \
+	tmuxinator_completion
```
Comment (JeroenBoersma) [inline Makefile:549]: i would make tmuxinator optional :)

**Comment by JeroenBoersma** [comment]
There are a few packages which I think are nice-to-have... (for instance tmuxinator brave(maybe not install them by default, because some of them add complexity too))

**Comment by JeroenBoersma** [comment]
I use https://github.com/zsh-users/antigen as a plugin manager for  zsh, I love it, but it's also personal flavor

**Comment by janmartenjongerius** [comment]
@JeroenBoersma your feedback is now covered in https://github.com/elgentos/setup/issues/1

**Pull request closed without merging**